### PR TITLE
connect mongo db and replaced mux routing with Gin

### PR DESCRIPTION
**Pull request by GordieH**
Added a make file so we can run a single command for testing. Also changed the mongo db .env info to match my local mongo set up.

### CODE REVIEW

**Review by VerteDinde — APPROVED** [review]
Nice!

File: go-server/main.go [hunk 1]
```diff
@@ -3,17 +3,13 @@ package main
...
-	//likely will switch to Gin or more direct net/http
+
 	r := router.Router()
-	// fs := http.FileServer(http.Dir("build"))
-	// http.Handle("/", fs)
+	log.Fatal(r.Run(":8080"))
 	fmt.Println("Starting server on the port 8080...")
-
-	log.Fatal(http.ListenAndServe(":8080", r))
 }
```
Comment (GordieH) [inline go-server/main.go:15]: just cleaned up the comments and calls

File: go-server/router/router.go [hunk 2]
```diff
@@ -2,20 +2,56 @@ package router
...
+			})
+		})
+	}
+
+	api.GET("/localreps", middleware.LocalRepsHandler)
+	api.GET("/task", middleware.GetAllTask)
+	api.POST("/task", middleware.CreateTask)
+	api.PUT("/task/:id", middleware.TaskComplete)
+	api.PUT("/undoTask/:id", middleware.UndoTask)
+	api.OPTIONS("/deleteTask/:id", middleware.DeleteTask)
```
Comment (GordieH) [inline go-server/router/router.go:51]: originally had "{id}" (mux syntax) and changed to ":id" -- I usually use keyword parameters, not in-path...

File: go-server/router/router.go [hunk 3]
```diff
@@ -2,20 +2,56 @@ package router
...
+	api.DELETE("/deleteTask/:id", middleware.DeleteTask)
+	api.OPTIONS("/deleteAllTask", middleware.DeleteAllTask)
```
Comment (GordieH) [inline go-server/router/router.go:53]: I had never used a "DELETE" request explicitly from browser, I didn't realize you need both a DELETE and OPTIONS methods.

Comment (VerteDinde) [reply go-server/router/router.go:53]: To be honest, I didn't either, I've only ever seen DELETE - OPTIONS is new to me 🤔 I wonder if this is specific to Go? Will read up on it

**Review by VerteDinde — APPROVED** [review]
Tested and everything is working well! 👍 I can run all of the API actions without hitting the CORS errors

File: go-server/router/router.go [hunk 4]
```diff
@@ -2,20 +2,56 @@ package router
...
+
+	// Setup route group for the API
+	api := r.Group("/api")
+	{
+		api.GET("/", func(c *gin.Context) {
+			c.JSON(http.StatusOK, gin.H{
+				"message": "pong",
+			})
+		})
+	}
```
Comment (VerteDinde) [inline go-server/router/router.go:44]: Oh, I think we maybe got some funky formatting here - is there a Go linter we could set up?

Comment (GordieH) [reply go-server/router/router.go:44]: ya good call. Fixing it in this next commit

**Pull request merged**